Move OUTPUT_MODE logic from rcS to rc.interface.

mcsauder opened the pull request:
Hi,

This PR migrates the remaining OUTPUT_MODE logic block from the rcS script and places it into rc.interface.  OUTPUT_MODE logic blocks in rc.interface have also been grouped together at the top of the script.  This PR is copy/paste from rcS into rc.interface, and copy/paste from the bottom of rc.interface higher up to group logic together.

Testing was conducted on a pixracer.  No observed changes to the startup script output result and a startup script output compare is below for convenience.

Let me know if you have questions on this PR.

-Mark

![image](https://user-images.githubusercontent.com/2497951/43157521-882efac2-8f3a-11e8-8568-cb0ccb11ce9d.png)
![image](https://user-images.githubusercontent.com/2497951/43157547-9b1230dc-8f3a-11e8-92a5-c0b0bc3a27aa.png)

bkueng commented on ROMFS/px4fmu_common/init.d/rc.interface:
```diff
@@ -36,6 +36,152 @@ then
+	fi
+fi
+
+if [ $OUTPUT_MODE == fmu -o $OUTPUT_MODE == io ]
```
Can you move this block after the next (`if [ $MIXER != none -a $MIXER != skip ]`), because the next block sets `OUTPUT_DEV` and this block uses it (`pwm failsafe -c ${PWM_OUT} -p ${FAILSAFE} -d ${OUTPUT_DEV}`)

mcsauder commented:
Thanks for catching that @bkueng , I just replaced that logic block back at the bottom of rc.interface where it was previously located.  I'll look for any remaining opportunities to refactor that logic.

The serial console output diff shows the same line differences as posted above, no changes between upstream and this PR.  Rebased and pushed up!  Thanks again.
